nengo/tests/test_transforms.py

```diff
@@ -6,6 +6,44 @@
 from nengo.exceptions import BuildError, ValidationError
 
 
+@pytest.mark.parametrize("x_mul", (1, 2, 3, 4))
+@pytest.mark.parametrize("k_size", (1, 2, 3, 4))
+@pytest.mark.parametrize("stride", (1, 2, 3, 4))
+@pytest.mark.parametrize("padding", ("same", "valid"))
+def test_convolution_shape(padding, stride, k_size, x_mul, rng, allclose):
```

Comment:
It's tricky, because all of them cover potential edge cases.

Probably the best way to reduce the number of tests is rather than covering all combinations of all cases, pick a smattering that together cover all the different values of each parameter (basically, do a random search instead of a grid search).

If we want to be thorough, we could even have the full test (all combinations) run as part of our "slow" tests that we run on release (though honestly, it's actually quite fast to go through even all 128 combinations).

Reply:
Yeah it's tricky because they don't actually take a lot of time, it's just the sheer number of tests that it generates in the log. Maybe putting the tests in a separate file would be sufficient? Then it's a bit more obvious that this group of tests is doing something more specific that "testing transforms".